### DISCUSSION TITLE
Does this web-ui automatically update the repository or the extensions?

### DISCUSSION THREAD

**Discussion by diomakka** [vladmandic/sdnext#1319] [Q&A]
For my use case I desperately need my setup to work 100% of the time, and I'm anxious of stuff automatically updating behind my back.
I've tried searching for any info about this but I can't find definitive answer.

If I just run the webui.bat will this:

Pull latest changes from this repo
update extensions (like controlnet)

**Comment by SoraEnkai** [comment]
While it doesn't pull the repo automatically (you need to run it with `--upgrade` for that).
You'll want to run with `--skip-update` in your case as it will update submodules (built-in extensions) by default.

Check `Install` part in readme has all the flags:

Setup options:
 `--use-ipex`                       Use Intel OneAPI XPU backend, default: False
  `--use-directml`                   Use DirectML if no compatible GPU is detected, default: False
  `--use-cuda`                       Force use nVidia CUDA backend, default: False
  `--use-rocm`                       Force use AMD ROCm backend, default: False
  `--skip-update`                    Skip update of extensions and submodules, default: False
  `--skip-requirements`              Skips checking and installing requirements, default: False
  `--skip-extensions`                Skips running individual extension installers, default: False
  `--skip-git`                       Skips running all GIT operations, default: False
  `--skip-torch`                     Skips running Torch checks, default: False
  `--reinstall`                      Force reinstallation of all requirements, default: False
  `--debug`                          Run installer with debug logging, default: False
  `--reset`                          Reset main repository to latest version, default: False
  `--upgrade`                        Upgrade main repository to latest version, default: False
  `--safe`                           Run in safe mode with no user extensions

**Reply by diomakka** [reply]
Great, thanks.

I had tried the` --skip-update` before because I did read that, but I got an error saying that command didn't exist. I wrongly assumed that those arguments weren't intended for launching but only for installing or something.

Turns out I still have an old version where the argument I'm looking for is called `--noupdate`
But yeah thanks, wouldn't have given that another try without your comment.
